lmb-700 | limit the start and endate input iwhen creating a new mandataris

JonasVanHoof opened the pull request:
## Description

When creating a new mandataris the start and end date need to be validated that they are in the bestuursperiode of the bestuursorgaan in tijd. Added a validation for the date field that with the help from our application context can valdiate of the value in the input field is correct.

## How to test

1. Create a mandataris and insert a date that is not in the current bestuursperiode. 
2. Add a date that is exact the start or end date of the bestuursorgaan bindingStart or einde date

## Links to other PR's

- https://github.com/lblod/app-lokaal-mandatenbeheer/pull/223

bfidlers reviewed (changes requested):
Looks mostly good, but I'm having a few issues:
1. The corrigeer fouten seems to error whenever I try to open it for a burgemeester
<img width="841" alt="Screenshot 2024-09-24 at 17 02 36" src="https://github.com/user-attachments/assets/361be507-bcdf-471c-a4f9-e8d25506cb1a">
2. The update state also needs an update, now (was already like this) it only checks if the start date is after the start date that was entered before, but you also need to check if it is not past the end of the bestuursperiode
<img width="759" alt="Screenshot 2024-09-24 at 17 05 51" src="https://github.com/user-attachments/assets/06b8f0d2-ad96-434b-abf3-ef1ea30ca2c4">

JonasVanHoof commented:
@bfidlers 

1. More than one bestuursorgaan was passed here and than it returned the value null, now it will always get the first one 
2. Indeed I forgot about this one.. 
 - changed the input from date picker to a dateInput
 - added a check that the madnataris start date cannot be out of the bestuursperiode
 -  update the error message so it is clear why the error is shown (in the past or after bestuursperiode einde)
 -  added a reference date as the user was not able to fill in the current date of the mandataris again ( when opening the modal with date 24-09-2024 and than changing it to 25-09-2024 and than back to 24-09-2024 this gave an error because of the hours where not correct)
 3. Other:
 - fixed the loading state of the pas aan button
 - check if date is also changed in HasChanges() method
 - renamed the method to _getApplicationContextMetaTtl_

bfidlers reviewed (approved):
Noticed a few issues with the corrigeer fouten, where it sometimes does not work, however I found the same issue on master (I think it has something to do with the fractie handling), logged a ticket for that https://binnenland.atlassian.net/jira/software/c/projects/LMB/boards/225/backlog?selectedIssue=LMB-800, but I think this PR does not cause this fault and otherwise it looks good.
